### PR TITLE
fix(plateau-lr): plateau warmup extends one extra epoch (+1 more)

### PR DESCRIPTION
**Pull request by andrewwhitecdw**
Small fixes in `scheduler/plateau_lr.py`:

#### fix: plateau warmup extends one extra epoch

**Fix:** Replace:
```
    def step(self, epoch, metric=None):
        if epoch <= self.warmup_t:
            lrs = [self.warmup_lr_init + epoch * s for s in self.warmup_steps]
            super().update_groups(lrs)
```

with:
```
    def step(self, epoch, metric=None):
        if epoch < self.warmup_t:
            lrs = [self.warmup_lr_init + epoch * s for s in self.warmup_steps]
            super().update_groups(lrs)
```

#### fix: plateau state_dict loses scheduler state

**Fix:** Apply patch:
```diff
--- a/scheduler/plateau_lr.py
+++ b/scheduler/plateau_lr.py
@@ -55,13 +55,15 @@ class PlateauLRScheduler(Scheduler):
         self.restore_lr = None
 
     def state_dict(self):
-        return {
-            'best': self.lr_scheduler.best,
-            'last_epoch': self.lr_scheduler.last_epoch,
-        }
+        state = super().state_dict()
+        state['base_scheduler'] = self.lr_scheduler.state_dict()
+        return state
 
     def load_state_dict(self, state_dict):
-        self.lr_scheduler.best = state_dict['best']
-        if 'last_epoch' in state_dict:
-            self.lr_scheduler.last_epoch = state_dict['last_epoch']
+        if 'base_scheduler' in state_dict:
+            self.lr_scheduler.load_state_dict(state_dict['base_scheduler'])
+            del state_dict['base_scheduler']
+        self.__dict__.update(state_dict)
```

### Files changed
- `scheduler/plateau_lr.py`

### CODE REVIEW

**Comment by andrewwhitecdw** [comment]
Closing per adversarial audit: branch contains off-scope or not-a-bug changes.

**Comment by andrewwhitecdw** [comment]
script closed on accident

**Comment by andrewwhitecdw** [comment]
Closing after the adversarial audit found the proposed change was not safe or complete enough for a small focused PR. A revised fix can be submitted separately. Thanks for understanding.

**Comment by andrewwhitecdw** [comment]
Reopening — the previous bulk-close was issued in error. No action needed from maintainers.
